src/digitsDFA.cpp

```diff
@@ -0,0 +1,33 @@
+#include "digitsDFA.h"
+#include "states.h"
+#include <utility>
+using namespace std;
+
+int error(char c) {
+	return DS_ERROR; // always go to error
+}
+
+int checkNonZero(char c) {
+	if('1' <= c && c <= '9') {
+		return DS_ACCEPT;
+	}
+	return DS_ERROR; // was not a digit character between 1 to 9 go to error
+}
+
+int checkDigits(char c) {
+	if('0' <= c && c <= '9') {
+		return DS_ACCEPT;
+	}
+	return DS_ERROR; // was not a digit character go to error
+}
+
+void DigitsDFA::initDFA() {
+	// initialization of the dfa
```

Comment:
Maybe stick to this commenting style, instead of doing to horizontally, then the line above